### PR TITLE
browser mock comparator functions

### PR DESCRIPTION
**Pull request by mgrybyk**
## Proposed changes

- all browser filter options support comparator function
- ignore case of `Content-Type` header

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have added proper type definitions for new commands (if appropriate)

## Further comments

mention: @rickschubert

### CODE REVIEW

File: packages/webdriverio/tests/utils/interception/devtools.test.js [hunk 1]
```diff
@@ -156,7 +156,7 @@ test('decodes base64 responses', async () => {
     const mock = new NetworkInterception('**/foobar/**')
     await NetworkInterception.handleRequestInterception(cdpClient, [mock])({
         request: { url: 'http://test.com/foobar/test.html' },
-        responseHeaders: [{ name: 'Content-Type', value: 'application/json' }]
+        responseHeaders: [{ name: 'content-Type', value: 'application/json' }]
```
Comment (rickschubert) [inline packages/webdriverio/tests/utils/interception/devtools.test.js:159]: I noticed that you changed the capitalisation of the header here. Given that all headers are not case-sensitive, should this be `content-type` then?

**Review by rickschubert — APPROVED** [review]
Wow, so fast, this is amazing!

File: packages/webdriverio/tests/utils/interception/devtools.test.js (hunk 1 above)
Comment (rickschubert) [inline packages/webdriverio/tests/utils/interception/devtools.test.js:159]: I assume this should be `content-type`, given that the case changed in the other place as well?

**Comment by rickschubert** [comment]
[Looking at these lines](https://github.com/webdriverio/webdriverio/blob/64b24e6e77e34e3b2af99231c38701015c4c99bb/packages/webdriverio/src/utils/interception/devtools.js#L38), it seems to me like only one of the comparator functions needs to return true. Could we change it so that all the comparator functions the user adds need to return true?

Looking at this block I am also wondering why it works, given that all the comparator functions return `false` when there is no filter expectation given. Or am I having a thinking error in my mind? I would have expected that it needs to look like below:

```js
                if (
                    !filterMethod(request.method, mock.filterOptions.method) ||
                    !filterHeaders(request.headers, mock.filterOptions.requestHeaders) ||
                    !filterHeaders(responseHeaders, mock.filterOptions.headers) ||
                    !filterRequest(request.postData, mock.filterOptions.postData) ||
                    !filterStatusCode(responseStatusCode, mock.filterOptions.statusCode)
                ) {
                    continue
                }
```

Additionally, I think that each of the filter functions should return `true` if no expectation is given.

File: packages/webdriverio/tests/utils/interception/devtools.test.js (hunk 1 above)
Comment (mgrybyk) [reply packages/webdriverio/tests/utils/interception/devtools.test.js:159]: It's test, wanted to have this one in mixed case

**Review by christian-bromann — APPROVED** [review]
👍

File: packages/webdriverio/tests/utils/interception/devtools.test.js (hunk 1 above)
Comment (rickschubert) [reply packages/webdriverio/tests/utils/interception/devtools.test.js:159]: Oh, of course, thank you!

**Pull request merged**